sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client_connection_async.py

```diff
@@ -2349,6 +2351,46 @@ def __GetBodiesFromQueryResult(result):
         # Query operations will use ReadEndpoint even though it uses POST(for regular query operations)
         request_params = _request_object.RequestObject(typ, documents._OperationType.SqlQuery)
         req_headers = base.GetHeaders(self, initial_headers, "post", path, id_, typ, options, partition_key_range_id)
+
+        # check if query has prefix partition key
+        cont_prop = kwargs.pop("containerProperties", None)
+        partition_key = options.get("partitionKey", None)
+        isPrefixPartitionQuery = False
+        partition_key_definition = None
+        if cont_prop:
+            cont_prop = await cont_prop()
+            pk_properties = cont_prop["partitionKey"]
+            partition_key_definition = PartitionKey(path=pk_properties["paths"], kind=pk_properties["kind"])
+            if partition_key_definition.kind != "MultiHash":
+                isPrefixPartitionQuery = False
```

Comment:
nit: This is the default value, can these 3 `if` conditions be simplified to

```
if type(partition_key) == list and len(partition_key_definition['paths']) != len(partition_key):
  isPrefixPartitionQuery = True
```

Reply:
Yes, but I will still need to check if it is multihash as there is one case where a single hash partition will be a list.

Reply:
Which is that case?

Reply:
I believe the partition key value is passed as an empty list when it is an empty partition key.